### PR TITLE
[5.2] Implement Redirector contract

### PR DESCRIPTION
**Pull request by rkgrep**
Another missing contract.

### CODE REVIEW

File: src/Illuminate/Foundation/Http/FormRequest.php [hunk 1]
```diff
@@ -5,7 +5,7 @@
 use Illuminate\Http\Request;
 use Illuminate\Http\Response;
 use Illuminate\Http\JsonResponse;
-use Illuminate\Routing\Redirector;
+use Illuminate\Contracts\Routing\Redirector;
```
Comment (lucasmichot) [inline src/Illuminate/Foundation/Http/FormRequest.php]: The imports must be sorted by string length, then alphabetically

**Comment by GrahamCampbell** [comment]
Please fix the phpdoc to match our cs.

**Comment by taylorotwell** [comment]
Unnecessary really. Why you want this?

**Comment by rkgrep** [comment]
There are only several things that don't have interfaces in [app](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Foundation/Application.php#L1028) and that gives hard dependency for specific classes (it's OK for managers however).

This is my personal opinion on how DI should work.

````php
app(RedirectorContract::class)->back();
````

If you don't like this, just close it together with #10884

**Pull request closed without merging**